lib/bolt/transport/ssh/connection.rb

```diff
@@ -63,15 +63,15 @@ def delete
         attr_reader :logger, :user, :target
         attr_writer :run_as
 
-        def initialize(target, transport_logger, load_config = true)
+        def initialize(target, transport_logger)
           # lazy-load expensive gem code
           require 'net/ssh'
           require 'net/ssh/proxy/jump'
 
           @target = target
-          @load_config = load_config
+          @load_config = target.options['load-config']
 
-          ssh_user = load_config ? Net::SSH::Config.for(target.host)[:user] : nil
+          ssh_user = @load_config ? Net::SSH::Config.for(target.host)[:user] : nil
```

Comment:
Can we handle this when we set `load_config`? can we handle the default in config instead of the initializer?

Reply:
Do you mean load the SSH config when we create the Target instance? We don't currently have any transport-specific code in the Target class, and this would mean adding some.

Reply:
I meant either 
just compute this a few lines earlier and make @load_config the computed value.
or
set the default as a normal config default.